docs/topics/native/native-improving-compilation-time.md

```diff
@@ -0,0 +1,69 @@
+  you might need to make the heap even larger or choose the right number of threads with `org.gradle.parallel.threads`.
+
+* **Build only the binaries you need**. Don't run Gradle tasks that build the whole project, such as `build` or `assemble`,
+  unless you really need to. These tasks build the same code multiple times, increasing the compilation time. In typical
+  cases such as running tests from IntelliJ IDEA or starting the app from Xcode, the Kotlin tooling avoids executing unnecessary
+  tasks. 
+  
+  If you have a non-typical case or build configuration, you might need to choose the task yourself.
+    * `linkDebug*`: To run your code during development, you usually need only one binary, so running the corresponding
+      `linkDebug*` task should be enough. Keep in mind that compiling a release binary (`linkRelease*`) takes more time than compiling a debug one
+    * `packForXcode`: Since iOS simulators and devices have different processor architectures, it's a common approach to
+      distribute a Kotlin Native binary as a universal (fat) framework. During local development, it will be faster to build
+      the `.framework` for only the platform you’re using. You can do this by calling the  `packForXcode` task generated
+      by the KMM project wizard. Remember that in this case, you will need to clean the build using `./gradlew clean` after
+      switching between the device and the simulator. See [this issue](https://youtrack.jetbrains.com/issue/KT-40907) for details.
+
+* **Don’t disable the [Gradle daemon](https://docs.gradle.org/current/userguide/gradle_daemon.html)** without having a
+  good reason to. [Kotlin/Native runs from the Gradle daemon](https://blog.jetbrains.com/kotlin/2020/03/kotlin-1-3-70-released/#kotlin-native)
+  by default. When it’s enabled, the same JVM process is used and there is no need to warm it up for each compilation.
+
+* **Use the Gradle [build caches](https://docs.gradle.org/current/userguide/build_cache.html)**:
+    * **Local build cache**: Add `org.gradle.caching=true` to your `gradle.properties` or run with `--build-cache` on the command line.
+    * **Remote build cache** in continuous integration environments. Learn how to [configure the remote build cache](https://docs.gradle.org/current/userguide/build_cache.html#sec:build_cache_configure_remote).
+
+* **Use the compiler caches**. Starting from 1.5.0-M1, `linuxX64` and `iosArm64` targets have experimental opt-in support
+  for compiler caches that improve compilation time for debug builds (for `linuxX64`, this feature is only available on Linux hosts).
+  You can use them by adding `kotlin.native.cacheKind.linuxX64=static` or `kotlin.native.cacheKind.iosArm64=static` to `gradle.properties`.
+
+  `iosX64` and `macosX64` targets already have the compiler caches enabled by default.
+
+* **Enable previously disabled features of Kotlin/Native**. If you have the `kotlin.native.disableCompilerDaemon=true` or
```

Comment:
Don't you think it's better to describe in this order:
1. There are some obsolete props.
2. Check them and remove them.

It seems more natural to me

Reply:
Not that they're obsolete.
It's like "compiler caches and daemon had issues in early versions, this is why we added properties that disable them. If you have these properties in your config, try removing them because issues you've faced may be already fixed."